### PR TITLE
Add documentation about pricing related to market orders

### LINKED ISSUE #3160: ROI wrongly triggered in Live env
## Step 1: Have you search for this issue before posting it?

If you have discovered a bug in the bot, please [search our issue tracker](https://github.com/freqtrade/freqtrade/issues?q=is%3Aissue). 
If it hasn't been reported, please create a new issue.

## Step 2: Describe your environment

  * Operating system: MAOS X
  * Python Version: 3.7.4
  * CCXT version: 1.24.83
  * Branch: Master 
  * Last Commit ID: 9abfdaaaa240dcab2fbd105982ed6eed025d8936
  
## Step 3: Describe the problem:
I'm facing a strange behaviour in the live environment (which is not happening in dry run). The bot is setup with "process_throttle_secs": 60,  "ticker_interval" : "5m",
minimal_roi = {
            "30": 0.005,
            "15": 0.01,
            "10": 0.02,
            "0":  0.03
        }.
Sometime like this morning, it manages to buy FET/BTC and sell it after 10sec with sell reason roi (and a loss of -0.66%).

### Relevant code exceptions or logs:

```
2020-04-13 07:20:26,159 - PriceFilter - INFO - Removed QLC/BTC from whitelist, because 1 unit is 0.794%
2020-04-13 07:21:23,768 - freqtrade.wallets - INFO - Wallets synced.
2020-04-13 07:21:23,770 - freqtrade.freqtradebot - INFO - Buy signal found: about create a new trade with stake_amount: 0.0005 ...
2020-04-13 07:21:23,770 - freqtrade.freqtradebot - INFO - Executing Buy for FET/BTC
2020-04-13 07:21:23,770 - freqtrade.freqtradebot - INFO - Getting price from order book Bid side.
2020-04-13 07:21:24,273 - freqtrade.freqtradebot - INFO - ...top 9 order book buy rate 0.00000210
2020-04-13 07:21:24,789 - freqtrade.freqtradebot - INFO - Found open order for Trade(id=None, pair=FET/BTC, amount=238.00000000, open_rate=0.00000219, open_since=closed)
2020-04-13 07:21:25,309 - freqtrade.persistence - INFO - Updating trade (id=None) ...
2020-04-13 07:21:25,309 - freqtrade.persistence - INFO - MARKET_BUY has been fulfilled for Trade(id=None, pair=FET/BTC, amount=238.00000000, open_rate=0.00000219, open_since=closed).
2020-04-13 07:21:25,814 - freqtrade.wallets - INFO - Wallets synced.
2020-04-13 07:21:26,319 - freqtrade.wallets - INFO - Wallets synced.
2020-04-13 07:21:26,322 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': buy, 'exchange': 'Binance', 'pair': 'FET/BTC', 'limit': 2.1899999999999998e-06, 'order_type': 'market', 'stake_amount': 0.0005212200000000001, 'stake_currency': 'BTC', 'fiat_currency': 'EUR', 'amount': 238.0, 'open_date': datetime.datetime(2020, 4, 13, 5, 21, 24, 789567), 'current_rate': 2.1e-06}
2020-04-13 07:21:27,322 - freqtrade.wallets - INFO - Wallets synced.
2020-04-13 07:21:33,696 - freqtrade.worker - INFO - Bot heartbeat. PID=55356, version='2020.3', state='RUNNING'
2020-04-13 07:21:34,508 - freqtrade.freqtradebot - INFO - Executing Sell for FET/BTC. Reason: SellType.ROI
2020-04-13 07:21:35,020 - freqtrade.wallets - INFO - Wallets synced.
2020-04-13 07:21:36,019 - freqtrade.persistence - INFO - Updating trade (id=42) ...
2020-04-13 07:21:36,020 - freqtrade.persistence - INFO - Marking Trade(id=42, pair=FET/BTC, amount=238.00000000, open_rate=0.00000219, open_since=closed) as closed as the trade is fulfilled and found no open orders for it.
2020-04-13 07:21:36,020 - freqtrade.persistence - INFO - MARKET_SELL has been fulfilled for Trade(id=42, pair=FET/BTC, amount=238.00000000, open_rate=0.00000219, open_since=closed).
2020-04-13 07:21:36,025 - freqtrade.freqtradebot - INFO - Getting price from order book Ask side.
2020-04-13 07:21:36,794 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': sell, 'exchange': 'Binance', 'pair': 'FET/BTC', 'gain': 'loss', 'limit': 2.18e-06, 'order_type': 'market', 'amount': 238.0, 'open_rate': 2.1899999999999998e-06, 'current_rate': 2.19e-06, 'profit_amount': -3.42e-06, 'profit_ratio': -0.00655509, 'sell_reason': 'roi', 'open_date': datetime.datetime(2020, 4, 13, 5, 21, 24, 789567), 'close_date': datetime.datetime(2020, 4, 13, 5, 21, 36, 20045), 'stake_currency': 'BTC', 'fiat_currency': 'EUR'}
```

### PR DESCRIPTION
**Pull request by xmatthias**
## Summary
Add documentation about pricing related to market orders

closes #3160

## Quick changelog

- Add documentation section explaining how pricing should be configured for market orders.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -553,6 +559,29 @@ A fixed slot (mirroring `bid_strategy.order_book_top`) can be defined by setting
...
+
+``` jsonc
+  "order_types": {
+    "buy": "market",
+    "sell": "market"
+    // ...
+  },
+  "bid_strategy": {
+    "price_side": "ask",
+    //...
```
Comment (hroff-1902) [inline docs/configuration.md:575]:
```suggestion
    // ...
```

File: docs/configuration.md [hunk 2]
```diff
@@ -553,6 +559,29 @@ A fixed slot (mirroring `bid_strategy.order_book_top`) can be defined by setting
...
+  },
+  "ask_strategy":{
+    "price_side": "bid",
+    //...
```
Comment (hroff-1902) [inline docs/configuration.md:579]:
```suggestion
    // ...
```

File: docs/configuration.md [hunk 3]
```diff
@@ -553,6 +559,29 @@ A fixed slot (mirroring `bid_strategy.order_book_top`) can be defined by setting
 
 When not using orderbook (`ask_strategy.use_order_book=False`), the price at the `ask_strategy.price_side` side (defaults to `"ask"`) from the ticker will be used as the sell price.
 
+### Market order pricing
+
+When using market orders, prices should be configured to use the "correct" side of the orderbook to allow realistic pricing detection.
```
Comment (hroff-1902) [inline docs/configuration.md:564]: "When using market orders with orderbook enabled, prices should be configured ..."?

Comment (xmatthias) [reply docs/configuration.md:564]: no? unless i misunderstand your intend.

it does not matter if you use orderbook or not, it's the side that matters.


ticker-prices (ask / bid values from the `fetch_ticker()` result do correspond (should correspond) to the best orderbook bid / ask prices.
The real difference between using ticker and orderbook is speed (at least i feel that way).

orderbook is usually a "cheaper" call with less impact on ratelimits (at least according to many exchange documentations) - but no matter if using ticker or orderbook - the price-side is applied and should be reversed for market orders.

Now for market orders, you could drive this configuration further and use the 2nd or 3rd slot in the oderbook assuming you're pushing past the first slot - but that'll too strongly depend on markets (and your stake-amount) - so i would rather not complicate this section too much.

Comment (hroff-1902) [reply docs/configuration.md:564]: >it does not matter if you use orderbook or not, it's the side that matters.

I just do not remember does it have effect when orderbook is not enabled (and ticker data is therefore used)

okay, not "with orderbook enabled", forget it...

**Comment by hroff-1902** [comment]
> should be reversed for market orders.

A side question: why cannot we deduct the needed side from the order type (buy limit -- take bid, buy market -- take ask, sell limit -- take ask, sell market: take bid)?...

**Comment by xmatthias** [comment]
> A side question: why cannot we deduct the needed side from the order type (buy limit -- take bid, buy market -- take ask, sell limit -- take ask, sell market: take bid)?...

we could in theory - but especially for limit orders, everyone has different preferences.

personally, i revert sides for limit buys as well - that way my unfilled rate / unfilled timeouts are close to 0. i do victimize spread for this (so a few 0.xx% of the trade) - but my strategy does not need this - and i get the fillability close to orders, but without the 700k BTC buying risk ... 

In the end though, the decision about which side to use (also for market orders) is up to the user.


What i thought we could do is warn if we detect market orders without sides reversed - so people are aware about this ... 
but i thought about this for months now - so i guess having this documented is a great first step :laughing:

**Comment by hroff-1902** [comment]
> What i thought we could do is warn if we detect market orders without sides reversed - so people are aware about this ...

Yes, that's nice, plus deducting the side from order type if no price_side is set (i.e. add default price_side).
As I guess...

**Comment by xmatthias** [comment]
no, i wouldn't do that (take the decision for the user) - there is one default for price_side per side ...

At maximum, we could make this required and force people to decide explicitly ...

**Comment by hroff-1902** [comment]
Why not? it would be logical and easy to explain: the "proper" side is used by default, which depends on the order type. If a user want to cross the spread -- he could specify it explicitly. Plus, for market orders, she could only chose the "proper" side.

I think the issue is caused by this non-logical behavior. I also remember similar questions on Slack...

Now we are forcing the user to make changes in two places when she wants to change limit orders to market orders or vice-versa. Which can be unnoticed and the wrong side be used. Especially if no "price_side" is specified explicitly and the default value is used...

At least, if you still think it's bad, let's make it non-optional.

**Pull request merged**